### PR TITLE
More spec binary updates

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

**Comment by kripken** [comment]
Ok, that should do it. Reading through the Binary Format document, I don't see anything that we are doing differently (except for postorder).

File: src/wasm-binary.h [hunk 1]
```diff
@@ -882,16 +907,11 @@ class WasmBinaryWriter : public WasmVisitor<WasmBinaryWriter, void> {
...
-          o << int8_t(BinaryConsts::I8Const) << uint8_t(value);
-          break;
-        }
-        o << int8_t(BinaryConsts::I32Const) << value;
+        o << int8_t(BinaryConsts::I32Const) << LEB128(curr->value.geti32());
         break;
       }
       case i64: {
-        o << int8_t(BinaryConsts::I64Const) << curr->value.geti64();
+        o << int8_t(BinaryConsts::I64Const) << LEB256(curr->value.geti64());
```
Comment (binji) [inline src/wasm-binary.h]: it's still LEB128. 128 refers to the base, not the length of the value.

File: src/wasm-binary.h [hunk 2]
```diff
@@ -882,16 +907,11 @@ class WasmBinaryWriter : public WasmVisitor<WasmBinaryWriter, void> {
     if (debug) std::cerr << "zz node: Const" << curr << " : " << curr->type << std::endl;
     switch (curr->type) {
       case i32: {
-        uint32_t value = curr->value.geti32();
-        if (value <= 255) {
-          o << int8_t(BinaryConsts::I8Const) << uint8_t(value);
-          break;
-        }
-        o << int8_t(BinaryConsts::I32Const) << value;
+        o << int8_t(BinaryConsts::I32Const) << LEB128(curr->value.geti32());
```
Comment (binji) [inline src/wasm-binary.h]: `i32.const` and `i64.const` should be sign-extended LEBs.

File: src/wasm-binary.h (hunk 1 above)
Comment (kripken) [inline src/wasm-binary.h]: Thanks! I should read docs more... ;)

File: src/wasm-binary.h (hunk 2 above)
Comment (kripken) [inline src/wasm-binary.h]: Thanks, I missed that.

**Pull request merged**